src/js/components/covid19/recipient/map/MapWrapper.jsx

```diff
@@ -376,20 +385,42 @@ export default class MapWrapper extends React.Component {
             return;
         }
 
+        // load the data source
         const source = mapboxSources[this.props.activeFilters.territory];
-        // calculate the range of data
-        const scale = calculateCovidMapRange(this.props.data.values, this.props.activeFilters.territory);
-        const colors = this.getColors();
+
+        // the d3 function takes a range array for the key/index for each quantile or segment in the case of a linear scale
+        let rangeArray = [];
+        let colors = [];
+        if (this.props.activeFilters.territory === 'state') {
+            colors = this.getColors(numStateQuantiles);
+            rangeArray = [...Array(numStateQuantiles).keys()];
```

Comment:
do we need range array to be an array of indices? or do we just need an array with the same length as the colors array? if the latter, then we can just use the colors.

Reply:
it needs to be an array of indices because the D3 scale functions map a domain to a range, so the rangeArray cannot be empty. The numbers in the range array [0, 1, 2....] correspond to to the layers that are added to the map. Similar to what we were doing when there were only 6 layers we used to color the map.